xilem/src/view/canvas.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use masonry::core::WidgetMut;
+use masonry::widgets;
+use vello::Scene;
+use vello::kurbo::Size;
+use xilem_core::MessageContext;
+
+use crate::core::{Mut, ViewMarker};
+use crate::{MessageResult, Pod, View, ViewCtx};
+
+/// Creates a non-interactive drawing surface.
+///
+/// The `canvas` function provides a way to render custom graphics using a user-
+/// supplied drawing callback. The callback receives a mutable reference to a
+/// `Scene` and the current allocated `Size` of the canvas, allowing you to draw
+/// shapes, images, or any other custom content.
+///
+/// # Example
+///
+/// ```ignore
+/// use xilem::view::canvas;
+/// use masonry::{Scene, Size};
+///
+/// let my_canvas = canvas(|scene: &mut Scene, size: Size| {
+///     // Drawing a simple rectangle that fills the canvas.
+///     scene.fill_rect((0.0, 0.0, size.width, size.height), [0.2, 0.4, 0.8, 1.0]);
+/// });
```

Comment:
I think we should strive towards not ignoring the examples in doc-comments.
This has the tendency to quickly be out of sync, I think you can already see that in the diff.

```suggestion
/// ```
/// use xilem::masonry::{kurbo::{Rect, Size}, peniko::Fill, vello::Scene};
/// use xilem::{Affine, view::canvas};
/// # use xilem::WidgetView;
///
/// # fn fill_canvas() -> impl WidgetView<()> + use<> {
/// let my_canvas = canvas(|scene: &mut Scene, size: Size| {
///     // Drawing a simple rectangle that fills the canvas.
///     scene.fill(
///         Fill::NonZero,
///         Affine::IDENTITY,
///         xilem::palette::css::AQUA,
///         None,
///         &Rect::new(0.0, 0.0, size.width, size.height),
///     );
/// });
/// # my_canvas
/// # }
```